.github/workflows/continuous-integration.yml

```diff
@@ -76,8 +78,16 @@ jobs:
         cmake . -LAH
         cmake --build . --config Release -- /maxcpucount:2 
 
+    - name: Cache opensim-core
+      id: cache-core
+      uses: actions/cache@v1
+      with:
+        path: ~/opensim-core-install
+        # https://help.github.com/en/actions/automating-your-workflow-with-github-actions/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+        key: ${{ runner.os }}-${{ hashFiles('opensim-core/*') }}
```

Comment:
What is the desired caching behavior? Right now, the cache is invalidated any time the master branch of opensim-core is updated. Another option is for this repo to contain a file that lists the desired commit/tag of opensim-core.

If this is the case, keep in mind that these actions should not be used to create formal releases.

Reply:
@chrisdembia It's a good point. As of now I'm using master to avoid unnecessary rebuilds when GUI changes but core doesn't, and also so that we detect when a change in core breaks the GUI build. At some point we need to switch to a specific commit for core which ideally is specified in a file/hash. Not sure yet how to set this up in github actions without modifying the yml file when we switch. Separate workflow?

Reply:
Thanks for clarifying; it would be nice to add a comment expressing your first point.

A few thoughts:
- We can add a plain-text file `opensim-gui/opensim-core-commit.txt` that we use to invalidate the cache for a pre-defined version of opensim-core.
- We could have a separate workflow that is a nightly build that checks to see if there is a new commit of `opensim-core`, and if so, uploads an updated artifact that the continuous-integration workflow downloads (using `actions/download-artifact`).
- When the continuous-integration script is not burdened with building opensim-core, it'll be snappier, and we could have an additional build that tests against a released version of opensim-core.